### PR TITLE
Feature optionnal product tv 211

### PR DESCRIPTION
**Pull request by willkoua**
| Q                                                      | R
| ------------------------------------------ | -------------------------------------------
| Type of contribution ?                      | [Features]
| Tickets (_issues_) concerned               | [211]

---

## What have you changed ?
add optionnal product feature

## How did you change it ?
...

## Screenshots
...

## Verification :
 -  [x] My branch name respect the standard defined in `CONTRIBUTING.md`
 -  [x] This Pull-Request fully meets the requirements defined in the issue
 -  [ ] I added or modified the attached tests

### CODE REVIEW

File: src/app/components/pages/admin/options-products/options-products-creation/options-products-creation.component.html [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    <multiselect [list]="listTypeOfProduct" [(selected)]="selectedTypeOfProduct" labelProperty="name">
+    </multiselect>
+  </div>
+
+  <div class="form-group">
+    <label class="nt-form-label">{{ 'coupons-creation.form.retreats' | translate }}</label>
+    <multiselect [list]="listRetreats"
+                 [(selected)]="selectedRetreats"
+                 labelProperty="name"
+                 (searchBar)="filterRetreats($event)"
```
Comment (Jerome-Celle) [inline src/app/components/pages/admin/options-products/options-products-creation/options-products-creation.component.html]: Une erreur apparaît quand on se sert du filtre, filterRetreats n'est pas une fonction définie.

File: src/assets/i18n/fr.json [hunk 2]
```diff
@@ -498,6 +504,16 @@
...
   "shared.form.coupon.value": "Valeur du coupon",
+  "shared.form.optionsproducts.name": "Nom",
+  "shared.form.optionsproducts.name_fr": "Nom en français",
+  "shared.form.optionsproducts.name_en": "Nom en anglais",
+  "shared.form.optionsproducts.url": "URL",
+  "shared.form.optionsproducts.price": "Price",
+  "shared.form.optionsproducts.details": "Details",
+  "shared.form.optionsproducts.details_fr": "Details en français",
+  "shared.form.optionsproducts.details_en": "Details en anglais",
+  "shared.form.optionsproducts.max_quantity": "quantité maximale",
```
[the remark this answers is not in the document]
Comment (willkoua) [reply src/assets/i18n/fr.json]: done

File: src/app/models/optionProduct.ts [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+  name: string;
+  name_en: string;
+  name_fr: string;
+  available: boolean;
+  price: string;
+  details: string;
+  details_fr: string;
+  details_en: string;
+  max_quantity: number;
+  order_lines: Array<string>;
```
[the remark this answers is not in the document]
Comment (willkoua) [reply src/app/models/optionProduct.ts]: done

File: src/app/components/pages/admin/options-products/options-products-creation/options-products-creation.component.html (hunk 1 above)
Comment (willkoua) [reply src/app/components/pages/admin/options-products/options-products-creation/options-products-creation.component.html]: done

**Pull request merged**